File: asc/semis_reports/report/user_performance_report/user_performance_report.py
# Copyright (c) 2013, Frappe Technologies and contributors
# For license information, please see license.txt

import frappe
from frappe.utils import cstr, cint, getdate, get_first_day, get_last_day, date_diff, add_days
from frappe import msgprint, _
from calendar import monthrange

def execute(filters=None):
	columns = get_columns(filters)
	conditions, filters = get_conditions(filters)
	data = get_data(conditions, filters)
	return columns, data

def get_data(conditions, filters):
	performance = frappe.db.sql("""SELECT 
									`tabASC`.owner, 
									u.full_name, 
									u.district,
									COUNT(`tabASC`.owner)
									FROM `tabASC`  inner join `tabUser` u 
									on `tabASC`.owner = u.name 
									WHERE `tabASC`.docstatus = 1 and `tabASC`.owner != 'Administrator'  %s 
									group by `tabASC`.owner"""% conditions, filters)
	return performance

def get_conditions(filters):
	conditions=""
	if filters.get("district"):
		conditions = "  and `tabASC`.district = %(district)s"
	return conditions, filters

def get_columns(filters):
	columns = [
		_("User ID") + "::120",
		_("User Name") + "::120",
		_("District") + ":Link/District:120",
		_("Created ASC") + "::120",
		]
	# columns.append({
    #         "label": _("District"),
    #         "fieldtype": "Link",
    #         "fieldname": "district",
    #         "options": "District",
    #         "width": 150
    # })
	return columns

File: asc/semis_reports/report/user_performance_report/user_performance_report.js
// Copyright (c) 2016, Frappe Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */
let district_by_user;
let read_only = false;
frappe.query_reports["User Performance Report"] = {
	"filters": [
		{
			"fieldname":"district",
			"label": __("District"),
			"fieldtype": "Link",
			"options": "District",
			"default":`${district_by_user}`
		}
	],
	onload:async function() {
		var system_manager = frappe.user_roles.includes('System Manager');
		if (system_manager == false) {
			console.log("if log");
			await frappe.call({
				method: "frappe.dirp.report.user_performance_report.user_performance_report.district_by_user",
				args: { user: frappe.session.user },
				freeze:true,
				callback: function (r) {
					district_by_user = r.message.for_value;
					var dictrict_filter = frappe.query_report.get_filter('district');
					dictrict_filter.df.default = district_by_user;
					dictrict_filter.refresh();
					dictrict_filter.set_input(dictrict_filter.df.default);
					$("input[data-fieldname='district']").attr('readonly', true);

				}
			});
		} else {
			district_by_user = '';
			var dictrict_filter = frappe.query_report.get_filter('district');
			dictrict_filter.df.default = district_by_user;
			dictrict_filter.refresh();
			dictrict_filter.set_input(dictrict_filter.df.default);
		}
},
};
